src/EnergyPlus/api/runtime.cc

```diff
@@ -75,6 +80,16 @@ void stopSimulation(EnergyPlusState state)
     thisState->dataGlobal->stopSimulation = true;
 }
 
+void setConsoleOutputState(EnergyPlusState state, int outputStatus)
+{
+    auto thisState = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
+    if (outputStatus == 0) {
+        thisState->dataGlobal->printConsoleOutput = false;
+    } else {
+        thisState->dataGlobal->printConsoleOutput = true;
+    }
+}
```

Comment:
In this new API function we take a 0 for no and a nonzero for yes to toggle on or off the console output.